httpserver/http_connection.cpp

```diff
@@ -806,22 +807,26 @@ void HttpClientSession::on_read(boost::system::error_code ec,
         return;
     }
 
-    init_callback(body);
+    init_callback(std::move(body));
 
     // If we get here then the connection is closed gracefully
 }
 
-void HttpClientSession::init_callback(std::shared_ptr<std::string> body) {
+void HttpClientSession::init_callback(std::shared_ptr<std::string>&& body) {
 
-    ioc_.post(std::bind(callback_, body));
+    sn_response_t res = {SNodeError::NO_ERROR, body};
```

Comment:
try `const sn_response_t res` here and below?

Reply:
Although it compiles, I feel weird about moving from a const object.

Reply:
feels weird, but you're not mutating the object itself! std::move just attempts to cast it as a rvalue ref, which works on a const...

Reply:
Move constructors/ assignment operators require non-const argument for T&&. I'm not sure why it works here, maybe it silently copies?

Reply:
So after looking into this a bit further, I think calling `foo(std::move(t))` with `void foo(T t)` will call the copy constructor `T(const T& other)` unless T has an explicit move constructor `T(T&& other)`, in which case the latter is called.

In our case, I guess we're getting a copy since we don't declare an explicit move constructor for `sn_response_t` ?

Reply:
Okay, I construct the response inline to avoid this argument ;)

Reply:
And you're right, the `const` might prevent the move constructor to intervene. So I am closing this discussion :)

Reply:
As far as I remember, special methods with move will be automatically generated as long as we don't declare our own special methods of any kind (which we don't).

Reply:
True! The rule of 3! Thanks for reminding me that, I didn't bother going that far.